test/lib/recordevents/event_info_store.go

```diff
@@ -173,7 +174,7 @@ func (ei *EventInfoStore) doRetrieveData() error {
 
 // Clean up any background resources used by the store.  Must be called exactly once after
 // the last use.
-func (ei *EventInfoStore) Cleanup() {
+func (ei *EventInfoStore) cleanup() {
 	close(ei.closeCh)
 }
```

Comment:
Can we keep this public so that we don't break downstream usages, and use `once.Do` to protect multiple `Cleanup` calls?

Reply:
The intention with this PR is to break downstream usages, so we make people aware of that change :smile: I also don't want to leave unused code around

Reply:
I'm gonna replicate this change to eventing-contrib too

Reply:
But now you break even people aware of that.

> I also don't want to leave unused code around

Why unused? We can use `ei.Cleanup` instead of `ei.cleanup` here: https://github.com/knative/eventing/pull/3780/files#diff-719035a5ad84fc383e0dea6ef4c67e32R96 and we make downstream usages happy.

Reply:
Ok let me try another solution

Reply:
that should do it